doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,231 @@
+.. _cmake_pkg:
+
+Zephyr CMake Package
+####################
+
+Zephyr can be exported as a CMake config package.
+
+A CMake config package allows a developed to integrate Zephyr into their own application by using
+``find_package(Zephyr)``.
+
+Exporting Zephyr as a CMake package is identical to create a reference to the current Zephyr
+installation inside the CMake user package registry.
+
+
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      In Linux, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: macOS
+
+      In macOS, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: Windows
+
+      In Windows, the CMake user package registry is found in:
+
+      ``HKEY_CURRENT_USER\Software\Kitware\CMake\Packages\Zephyr``
+
+
+The Zephyr package allows CMake to automatically find a Zephyr base.
+At minimum, a single Zephyr installation must be exported, but it is allowed to export multiple.
+Exporting multiple Zephyr installations may be useful when developing applications out-of-tree.
+
+
+Zephyr CMake package export (west)
+**********************************
+
+When installing Zephyr using :ref:`west <get_the_code>` then it is recommended to export Zephyr
+using ``west zephyr-export``.
+
+
+Zephyr CMake package export (without west)
+******************************************
+
+Zephyr can be exported using CMake directly.
+To do so, run the following command:
+
+``cmake -S <PATH-TO-ZEPHYR>/share/zephyr-package/cmake -B <PATH-TO-ZEPHYR>/share/zephyr-package/cmake``
+
+
+this will export the current Zephyr to CMake user package registry.
+
+.. _zephyr_cmake_package_zephyr_base:
+
+Zephyr Base Environment Setting
+*******************************
+
+The Zephyr CMake package search functionality allows for explicitly specifying a Zephyr base using
+an environment variable.
+
+In order to support the use of ``ZEPHYR_BASE`` environment setting, the following ``find_package()``
+syntax much be used:
+
+  .. code-block:: cmake
+
+    find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
+
+This syntax instructs CMake to first search for Zephyr at the location pointed to by ``$ZEPHYR_BASE``
+and if that is unset, then use the normal search pattern.
+
+.. _zephyr_cmake_search_order:
+
+Zephyr CMake Package Search Order
+*********************************
+
+When ``ZEPHYR_BASE`` is not used for searching, then the Zephyr matching the following criteria will
+be used
+
+* Zephyr in-tree: This means a sample located inside an existing tree, will automatically use the Zephyr in same tree.
+  As example::
+
+      <path-to>/zephyr-worktree/zephyr
+      <path-to>/zephyr-worktree/zephyr/samples/hello_world
+
+  in this example, ``hello_world`` will use ``<path-to>/zephyr-worktree/zephyr`` cause the Zephyr
+  and the application shares the same tree.
+
+
+* Zephyr in-working-tree: This means a sample located in a project tree, will use the Zephyr found in same work-tree.
```

Comment:
Does working-tree mean "west workspace"? If so, can you use that term? If not, I don't know what this means, can you please explain? :D

Reply:
If using `west` then it is the west workspace, but as users are allowed to use Zephyr without west, and the fact that Zephyr package does not rely on anything from west, I would be careful to not put west in the name, as it could give wrong impression.

How about `workspace` or `zephyr-workspace`, as opposed to zephyr-repo. ?
See also other comment.

Reply:
When not using west, I don't know what a working-tree is at all, then :(. Sorry, I'm even more confused now. I'm missing something.